### PR TITLE
pybind/rados: add Ioctx.get_pool_id() and Ioctx.get_pool_name()

### PR DESCRIPTION
**Pull request by zy751713126**
Signed-off-by: Zheng Yin <zhengyin@cmss.chinamobile.com>

### CODE REVIEW

File: src/pybind/rados/rados.pyx [hunk 1]
```diff
@@ -3192,6 +3194,43 @@ returned %d, but should return zero on success." % (self.name, ret))
         self.require_ioctx_open()
         return SnapIterator(self)
 
+    def get_id(self):
+        """
+        Get pool id
+
+        :returns: str - pool id
```
Comment (tchaikov) [inline src/pybind/rados/rados.pyx]: i don't think the type of the return value is `str`. see https://github.com/ceph/ceph/blob/aaad7dfc0b61c9023fc3295a151448a649173f03/src/include/rados/librados.h#L1003

File: src/pybind/rados/rados.pyx [hunk 2]
```diff
@@ -3192,6 +3194,43 @@ returned %d, but should return zero on success." % (self.name, ret))
         self.require_ioctx_open()
         return SnapIterator(self)
 
+    def get_id(self):
```
Comment (tchaikov) [inline src/pybind/rados/rados.pyx]: i'd suggest use `get_pool_id()`. as id in this context could be ambiguous.

File: src/pybind/rados/rados.pyx [hunk 3]
```diff
@@ -3192,6 +3194,43 @@ returned %d, but should return zero on success." % (self.name, ret))
...
+        """
+        Get pool id
+
+        :returns: str - pool id
+        """
+        with nogil:
+            ret = rados_ioctx_get_id(self.io)
+        return ret;
+
+    def get_name(self):
```
Comment (tchaikov) [inline src/pybind/rados/rados.pyx]: s/get_name/get_pool_name/

File: src/test/pybind/test_rados.py [hunk 4]
```diff
@@ -384,6 +384,9 @@ def test_obj_xattrs(self):
             stored_xattrs[key] = value
         eq(stored_xattrs, xattrs)
 
+    def test_get_name(self):
+        eq(self.ioctx.get_name(), 'test_pool')
```
Comment (tchaikov) [inline src/test/pybind/test_rados.py]: could add a test for `get_id()` or `get_pool_id()` using `Rados.pool_lookup()`.

File: src/pybind/rados/rados.pyx (hunk 2 above)
Comment (zy751713126) [reply src/pybind/rados/rados.pyx]: done

File: src/test/pybind/test_rados.py (hunk 4 above)
Comment (zy751713126) [reply src/test/pybind/test_rados.py]: I will add a test for get_pool_id, thanks a lot.

File: src/pybind/rados/rados.pyx (hunk 3 above)
Comment (zy751713126) [reply src/pybind/rados/rados.pyx]: done

File: src/pybind/rados/rados.pyx (hunk 1 above)
Comment (zy751713126) [reply src/pybind/rados/rados.pyx]: done

**Comment by zy751713126** [comment]
retest this please.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-08-15_06:56:12-rados-wip-kefu-testing-2019-08-15-1125-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-08-16_05:52:53-rados-wip-kefu-testing-2019-08-16-0928-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-08-16_11:16:42-rados-wip-kefu-testing-2019-08-16-1501-distro-basic-smithi/

**Pull request merged**